ngraph/core/src/op/binary_convolution.cpp

```diff
@@ -73,80 +73,119 @@ op::v1::BinaryConvolution::BinaryConvolution(const Output<Node>& data,
 void op::v1::BinaryConvolution::validate_and_infer_types()
 {
     NGRAPH_OP_SCOPE(v1_BinaryConvolution_validate_and_infer_types);
-    const PartialShape& data_batch_shape = get_input_partial_shape(0);
+    const PartialShape& data_batch_pshape = get_input_partial_shape(0);
     element::Type data_batch_et = get_input_element_type(0);
-    const PartialShape& filters_shape = get_input_partial_shape(1);
+    const PartialShape& filters_pshape = get_input_partial_shape(1);
 
-    PartialShape result_shape = PartialShape::dynamic();
-    if (data_batch_shape.rank().is_static())
-    {
-        result_shape =
-            std::vector<Dimension>(data_batch_shape.rank().get_length(), Dimension::dynamic());
+    NODE_VALIDATION_CHECK(this,
+                          data_batch_et.is_real(),
+                          "Data batch element type must be float point. Got: ",
+                          data_batch_et);
 
-        if (data_batch_shape.rank().get_length() > 1)
-        {
-            result_shape[0] = data_batch_shape[0]; // batch size
-        }
+    // TODO: Add NodeValidationCheck to filters et once u1 is supported in nGraph Python API
```

Comment:
Please add a ticket number in the comment.

Reply:
Done